LDAR_Sim/model_code/generic_functions.py

```diff
@@ -245,3 +247,32 @@ def check_ERA5_file(wd, target_file):
             print("Authentication Failed or Server Unavailable. Exiting")
             sys.exit()
         print("Weather data download complete")
+
+def read_parameter_file (filename, verbose = True):
```

Comment:
I had some troubles loading the file because the function uses the current working directory.  The following could be pushed in to get the relative parent directory:
`src_dir_path = Path(os.path.dirname(os.path.realpath(__file__)))
 src_dir = str(src_dir_path)
 root_dir = str(src_dir_path.parent)`

here src_dir  is **always** "..../LDAR_Sim/module_code" and root_dir is **always** "..../LDAR_Sim/module_code" regardless of the projects working directory.

Reply:
I don't follow? Should the generic read_parameter_file function always look for parameter filenames in a certain directory?
I think this function can handle any file path, either absolute or relative to the working directory. Same deal with any filename in any parameter file.
Maybe there is a more global enhancement to more comprehensively support putting both input and output files anywhere, not always relative to the model code?
Apologies, I'll take a deeper look at this later this week.